x-pack/plugins/secops/public/containers/ip_overview/index.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { getOr } from 'lodash/fp';
+import React from 'react';
+import { Query } from 'react-apollo';
+import { pure } from 'recompose';
+
+import { GetIpOverviewQuery, IpOverviewData } from '../../graphql/types';
+import { networkModel } from '../../store/local';
+import { createFilter } from '../helpers';
+import { QueryTemplateProps } from '../query_template';
+
+import { ipOverviewQuery } from './index.gql_query';
+
+export interface IpOverviewArgs {
+  id: string;
+  ipOverviewData: IpOverviewData;
+  loading: boolean;
+}
+
+export interface IpOverviewProps extends QueryTemplateProps {
+  children: (args: IpOverviewArgs) => React.ReactNode;
+  type: networkModel.NetworkType;
+  ip: string;
+}
+
+export const IpOverviewQuery = pure<IpOverviewProps>(
+  ({ id = 'ipOverviewQuery', children, filterQuery, sourceId, ip }) => (
+    <Query<GetIpOverviewQuery.Query, GetIpOverviewQuery.Variables>
+      query={ipOverviewQuery}
+      fetchPolicy="cache-and-network"
+      notifyOnNetworkStatusChange
+      variables={{
+        sourceId,
+        filterQuery: createFilter(filterQuery),
+        ip,
+      }}
+    >
+      {({ data, loading }) => {
+        const ipOverviewData = getOr({}, `source.IpOverview`, data);
```

Comment:
I do not expect you to change anything here but do want to point out two extra lines if you want to use them for more TypeSafety.

```ts
const init: IpOverviewData = {};
const ipOverviewData: IpOverviewData = getOr(init, `source.IpOverview`, data);
```

You don't have to use that, but if you do, and later something becomes required on ipOverviewData your typescript check will catch it. At the moment, `getOr({}` will take any for the first argument and likewise if you look at the type it returns `any` as well.

<img width="511" alt="Screen Shot 2019-04-01 at 4 51 18 PM" src="https://user-images.githubusercontent.com/1151048/55364580-6bb37600-549e-11e9-8dcc-721e89bc80f4.png">

I tried to use `getOr` before with templates but couldn't get it to work like JavaScript arrays. That syntax above this was the cleanest I could get my safety.

Reply:
Oh this is great -- thanks! :)